test/integration/controller/jobs/raycluster/raycluster_controller_test.go

```diff
@@ -0,0 +1,748 @@
+
+		cfg = fwk.Init()
+		ctx, k8sClient = fwk.RunManager(cfg, managerSetup(jobframework.WithManageJobsWithoutQueueName(true)))
+	})
+	ginkgo.AfterAll(func() {
+		fwk.Teardown()
+	})
+
+	var (
+		ns          *corev1.Namespace
+		wlLookupKey types.NamespacedName
+	)
+	ginkgo.BeforeEach(func() {
+		ns = &corev1.Namespace{
+			ObjectMeta: metav1.ObjectMeta{
+				GenerateName: "core-",
+			},
+		}
+		gomega.Expect(k8sClient.Create(ctx, ns)).To(gomega.Succeed())
+
+		wlLookupKey = types.NamespacedName{Name: workloadraycluster.GetWorkloadNameForRayCluster(jobName), Namespace: ns.Name}
+	})
+	ginkgo.AfterEach(func() {
+		gomega.Expect(util.DeleteNamespace(ctx, k8sClient, ns)).To(gomega.Succeed())
+	})
+
+	ginkgo.It("Should reconcile RayClusters", func() {
+		ginkgo.By("checking the job gets suspended when created unsuspended")
+		priorityClass := testing.MakePriorityClass(priorityClassName).
+			PriorityValue(priorityValue).Obj()
+		gomega.Expect(k8sClient.Create(ctx, priorityClass)).Should(gomega.Succeed())
+
+		job := testingraycluster.MakeJob(jobName, ns.Name).
+			Suspend(false).
+			WithPriorityClassName(priorityClassName).
+			Obj()
+		err := k8sClient.Create(ctx, job)
+		gomega.Expect(err).To(gomega.Succeed())
+		createdJob := &rayclusterapi.RayCluster{}
+
+		setInitStatus(jobName, ns.Name)
+
+		gomega.Eventually(func() bool {
+			if err := k8sClient.Get(ctx, types.NamespacedName{Name: jobName, Namespace: ns.Name}, createdJob); err != nil {
+				return false
+			}
+			return false
+		}, util.Timeout, util.Interval).Should(gomega.BeFalse())
+
+		ginkgo.By("checking the workload is created without queue assigned")
+		createdWorkload := &kueue.Workload{}
+		gomega.Eventually(func() error {
+			return k8sClient.Get(ctx, wlLookupKey, createdWorkload)
+		}, util.Timeout, util.Interval).Should(gomega.Succeed())
+		gomega.Expect(createdWorkload.Spec.QueueName).Should(gomega.Equal(""), "The Workload shouldn't have .spec.queueName set")
+		gomega.Expect(metav1.IsControlledBy(createdWorkload, createdJob)).To(gomega.BeTrue(), "The Workload should be owned by the Job")
+
+		ginkgo.By("checking the workload is created with priority and priorityName")
+		gomega.Expect(createdWorkload.Spec.PriorityClassName).Should(gomega.Equal(priorityClassName))
+		gomega.Expect(*createdWorkload.Spec.Priority).Should(gomega.Equal(priorityValue))
+
+		ginkgo.By("checking the workload is updated with queue name when the job does")
+		jobQueueName := "test-queue"
+		createdJob.Annotations = map[string]string{constants.QueueAnnotation: jobQueueName}
+		gomega.Expect(k8sClient.Update(ctx, createdJob)).Should(gomega.Succeed())
+		gomega.Eventually(func() bool {
+			if err := k8sClient.Get(ctx, wlLookupKey, createdWorkload); err != nil {
+				return false
+			}
+			return createdWorkload.Spec.QueueName == jobQueueName
+		}, util.Timeout, util.Interval).Should(gomega.BeTrue())
+
+		// no kind is registered for the type v1.RayCluster in scheme "pkg/runtime/scheme.go:100" for line 161
```

Comment:
You can uncomment this test case and update go modules for this branch to point to master. The PR will be blocked anyways until kuberay v1.1.0 is released anyways